shell/browser/web_contents_zoom_controller.cc

```diff
@@ -268,11 +269,18 @@ void WebContentsZoomController::ResetZoomModeOnNavigationIfNeeded(
   double old_zoom_level = zoom_map->GetZoomLevel(web_contents());
   double new_zoom_level = zoom_map->GetZoomLevelForHostAndScheme(
       url.scheme(), net::GetHostOrSpecFromURL(url));
-  event_data_ = std::make_unique<ZoomChangedEventData>(
+
+  ZoomChangedEventData zoom_change_data(
       web_contents(), old_zoom_level, new_zoom_level, false, ZOOM_MODE_DEFAULT);
+  for (auto& observer : observers_) {
+    observer.OnZoomChanged(zoom_change_data);
```

Comment:
What is the reason for this change ? Shouldn't the `updateState` in `DidFinishNavigation` take care of calling the observers ?

Reply:
oops, sorry, this should have been deleted!

Reply:
@deepak1556 that's what this fix is - removing `UpdateState`. It's what causes this bug.

Reply:
Can you please expand a bit more on the change. I am failing to understand how the call to observers here is different from `updateState` making the same call at the [end of the navigation cycle](https://github.com/electron/electron/blob/96f5f89f1bcb3aecb6786339cee92e26e651f6ba/shell/browser/web_contents_zoom_controller.cc#L390-L397). Is there a zoom level change in `SetZoomFactorOnNavigationIfNeeded` between these callsites which prompted this change ?

Reply:
@deepak1556 the codepath you linked in DidFinishNavigation is only hit in the case that `! navigation_handle->IsSameDocument` - the call to UpdateState i removed was happening on all navigations and that was the issue

Reply:
@deepak1556 done!